monome/teletype
# Add SCRIPT.POL op for changing script execution trigger polarity

#### What does this PR do?

Adds a `SCRIPT.POL s p` get/set op, with alias `$.POL`. When `s` is 1-8, `$.POL s p` will set script input `s` to fire with polarity `p`, where 1 is a rising edge (default), 2 is a falling edge, 3 is both edges, 0 is neither. Active edges are indicated in Live mode by highlighting different pixels in the mutes icon. `$.POL 0 p` will change all 8 script inputs at once.

#### Provide links to any related discussion on [lines](https://llllllll.co/).

[1](https://llllllll.co/t/teletype-measuring-gate-length/8212), [2](https://llllllll.co/t/teletype-3-feature-requests-and-discussion/16219/345)

#### How should this be manually tested?

Patch a clock to trigger input 1. In script 1, use `TR.TOG 1`. In live mode, use `$.POL 1 3` to change script 1 to "both" polarity, and observe trigger output 1 follow the clock. The mutes icon will now have another pixel to the right of the indicator for the first script lit, to indicate that both edges are active. Run `$.POL 1 2` to change this to only fire on a falling edge, and the first pixel in the mutes icon turns off.

#### Any background context you want to provide?

#### If the related Github issues aren't referenced in your commits, please link to them here.

#### I have,
